src/amber/pipes/client_ip.cr

```diff
@@ -0,0 +1,34 @@
+module Amber
+  module Pipe
+    # Handler to determine real client IP address when Amber is behind a
+    # reverse proxy and save it to context.client_ip.
+    #
+    # The proxy should pass client IP in a header such as "X-Forwarded-For".
+    # In case of multiple values in a header, the first one is used.
+    #
+    # Crystal does not currently make remote IP available, so it is impossible
+    # to accept the header value only when coming from trusted proxy IPs.
+    # For that and other benefits, it is suggested to always run Amber apps
+    # behind proxies such as HAProxy or Nginx.
+    class ClientIp < Base
+      def initialize(header : String = "X-Forwarded-For")
+        @headers = [header]
+      end
+      def initialize( @headers : Array(String))
+      end
+
+      def call(context : HTTP::Server::Context)
+        @headers.each do |header|
+          if addresses = context.request.headers.get?(header)
+            context.client_ip = Socket::IPAddress.new(addresses[0], 0)
```

Comment:
The `x-forwarded-proto` is similarly important (and risky). Should it be included here as well?

Reply:
Could you clarify please? From what I see, Amber doesn't consult x-forwarded-proto header anywhere, so it seems that there is no risk?
Are you simply suggesting to add the value of x-forwarded-proto to Context as well?

Reply:
Hm, that's interesting. I suppose my app is redirecting to https by nginx and not Amber. It's possible that Amber isn't doing anything with `x-forwarded-proto` at all. Since that seems to be the case, doing anything with it should be done in a separate PR. Carry on!